crates/nargo_cli/build.rs

```diff
@@ -119,51 +84,81 @@ fn compile_success_empty_{test_name}() {{
         .and(predicate::str::contains("| 0            |")));
 }}
             "#,
-            test_dir = test_dir.display(),
-        )
-        .expect("Could not write templated test file.");
-    }
-}
-
-fn generate_compile_success_contract_tests(test_file: &mut File, test_data_dir: &Path) {
-    let test_sub_dir = "compile_success_contract";
-    let test_data_dir = test_data_dir.join(test_sub_dir);
-
-    let test_case_dirs =
-        fs::read_dir(test_data_dir).unwrap().flatten().filter(|c| c.path().is_dir());
+                test_dir = test_dir.display(),
+            )
+            .expect("Could not write templated test file.");
```

Comment:
It looks like the only parts that differ between each of these generator functions is the raw string that is concatenated in each, is this correct? If so, we can forego the functions altogether and just pass in the strings directly into `generate_tests`.

Reply:
Good point! I removed the functions and I am now just passing the strings directly in!